src/components/changeUserData/ChangeEmail.tsx

```diff
@@ -0,0 +1,139 @@
+import * as Yup from 'yup';
+import axios from 'axios';
+import { Dispatch, SetStateAction, useContext } from 'react';
+import { ToastContentProps, toast } from 'react-toastify';
+import { useForm } from 'react-hook-form';
+import { useParams } from 'react-router-dom';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+import classes from '../../pages/Form.module.css';
+import { AuthContext } from '../../context/auth-context';
+
+interface UserFormValues {
+  email: string | null;
+}
+const UserFormSchema = () =>
+  Yup.object({
+    email: Yup.string()
+      .email('Invalid email.')
+      .required('Your email is required.'),
+  });
+
+const ChangeEmail = ({
+  isEditingEmail,
+  setIsEditingEmail,
```

Comment:
Add userDataEmail as prop, and use as value, same thing for name.

![image](https://github.com/KatarzynaChmielecka/space/assets/23500051/d4751224-624b-4c51-becf-b69c1d0ff2c5)